### DISCUSSION TITLE
Welcome to EPEM-Ethernet-Power-Energy-Monitor Discussions!

### DISCUSSION THREAD

**Discussion by DitroniX** [DitroniX/EPEM-Ethernet-Power-Energy-Monitor#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by DitroniX** [comment]
**EPEM** follows in the successful development footsteps of the IPEM and IPEC Power Energy Monitoring Boards.

**EPEM is an Ethernet Power Energy Monitor and Controller**. 3 Phase, or 3 x Single Phases, with DAC/PWM/MPWM/MPPT/I2C Expansion Port Outputs.

This is very much in development and work in progress (July 2023)

_Please feel free to contact me with any wish list ideas that the community would like to see in the design._

**Comment by guido999** [comment]
Hi Dave,
I'm new to ESP32 and hoping to finish my project nevertheless with some help from others (you?) and my friend 'euria'-ai.  I'm also not a electronics or IT expert LOL. But I have been using Home Assistant for several years now.
My goal is to do 3-phase power measurements, and also Thermocouple type-K measurements and bring them to Home Assistant. This is to control my Water Heater. This heater used to be 3-phase but with some relais I have made it such that it can go from 1 phase (plus nutral) three elements in series, two elements in series, two phases in thee-phase arrangement and three phases in thrwe-phase arrangement. This provides me with the ability to choose the power in steps from 880W, 1320W, 3750W, and 7500W which I intend to use for putting my excess solar power into the water heater.
Now I think I could use the EPEM board which is brilliant! But I have a few questions:
I intend to use Zigbee as I have a network already and it uses very little power. 
- is the MHF3-antenna for Wifi, Zigbee, or both?
- can I disable Ethernet and Wifi with jumpers or in the firmware to preserve power?
- do you intend to still make the non-ethernet/nin-wifi version which could be cheaper?

For the thermocouple I intend to use type K because I have very limited space in the heater-temperature tube. There is also the NTC from the heater thermostat and also the upper temp. limiting. And I don't want to take those out. A thermocouple will fit though.
Therefore I need a MAX31855. Which will need: CS, SCK, and SO/MISO pins available.  And VCC en GND from the 3.3V and GND headers.
- are these pins available and accessible for me to solder on the appropriate wires?
- I couldn't find it in your shop, can you provide a MAX31855?

Do you have any other thoughts on this project which could help me?

**Comment by DitroniX** [comment]
Hi Guido,

Many thanks for your question.

Regarding the 3-phase power measurements,  based on your explanation, you are single phase but split over three elements?   It is these elements you wish to monitor?

- On the EPEM, MHF3 is used for WiFI, Zigbee and Thread.
- Ethernet and all radio functions are disabled by default, you enable them in your firmware.

The cost difference to add/remove Ethernet is not much.   IPEM has sold Worldwide and is a reliable board (but needed replacement due to obsolescence on some parts, plus a better ESP), and a new version(s), are in the process of design.   

[IPEM Plus](https://github.com/DitroniX/IPEM-Plus-ESP32-C5-LoRa-LoRaWAN-IoT-Mains-Power-Energy-Monitor) (ATM90E36, C6 and LoRa), is the first, but another is in the cooking pot - which would effectively become the new IPEM (ATM90E32 with No Ethernet or LoRa).   This would reduce the cost slightly.  It may also have six CT inputs but that is not yet firmed up.

Ref: Thermocouple.  The Waterproof DS18B20-Compatible Temperature Sensors work very well and would recommend you look at these too.  I use these on my tank, as do others, at different water levels, and they work great to provide automation on heat storage.

You could use an I2C chip?

The IPEM Plus exposes a spare SPI port directly from the header.  EPEM does not but you could modify a board, which is not the ideal.  The new IPEM board (not yet released), does also have a SPI header.

All information on the boards are on GitHub.  Each repository (board), has its own dedicated Wiki pages.

[Comparison Chart - which maybe of Interest](https://github.com/DitroniX/IPEM-Plus-ESP32-C5-LoRa-LoRaWAN-IoT-Mains-Power-Energy-Monitor/wiki/Compare-Mains-Power-IoT-Energy-Monitors)

Another option which some users have done, it is use a [IPEM PiHat Lite](https://github.com/DitroniX/IPEM-PiHat-IoT-Power-Energy-Monitor), with an [ESPuno Pi Zero](https://github.com/DitroniX/ESPuno-Pi-Zero_WiFi-BT-Zigbee-Thread-60V
).  This works really well, and in fact I used this a lot in testing and gives you SPI as well.   The IPEM PiHat also has ADC in, so an option for NTC inputs?.

<img width="1000" alt="IPEM PiHat with ESPuno Pi Zero" src="https://github.com/user-attachments/assets/e733334e-f538-428e-a297-cde15cbcfa6b" />



**Question**: 

All historical boards had 3.5mm sockets for plugging in CT's.  With adding Rogowski options and the DIN mounting, terminal blocks are the better option, with [board links](https://github.com/DitroniX/IPEM-Plus-ESP32-C5-LoRa-LoRaWAN-IoT-Mains-Power-Energy-Monitor/wiki/Jumper-Configurations) to select the require option.

Still on the fence over which is the better options:  a) 3.5mm or b) Terminal blocks.    

What do you think?

A lot of info, but hope this is of help.

KR
Dave

**Comment by guido999** [comment]
Hi Dave,
thank you for your swift reply!
It is just singe phase in two out of four settings. I know, it's an unusual arrangement ;-).  I have it working with a dropdown in HA where I can choose the set power level 0/882/1323/3750/7500. The last two are three-phase star-configuration. I want to measure the three phases separately. In the first two power settings only phase 3 will be live, in modus '3750W' only phase 1 and 2 will be live, and in the highest setting all 3 phases will be live.
The reason I want to know the temperature is that if it hasn't been high enough for two weeks, I need to bring it up once for legionella protection. But if it's 85°C or higher, I need to set the valves such that the water won't pass the gas fired heater because it cannot handle that high input temperature. The gas boiler is still in series in case of cloudy days.
The thermo-sensor you mention looks a bit like a NTC element? These tend to be too big. The water heater has a measurement tube of about 5mm inner diameter, but there is already a sensor in for the max temp-prevention and also for for the thermostat of the water heater, both of which I want to keep in place as a safety measure. My electrical temperature measurement should be totally redundant. I now have a Thermocouple type K that fits neatly into the tube and it goes almost all the way up to the top of the tube. But my current measurement device is a Eurotherm 2132 which is not capable of any communication to Home Assistant.
The arrangement will come in a closed, water tight switch box which is already heating up a litle because of all the zigbee modules and relais. I don't want to add too much heat to that, so I would prefer just ESP and not a Pi.
If I do solder the MAX31855 onto the EPEM, where on the picture would that be? Is there enough space for a lumber jacker to solder it? 8-)

About the connection of the CT's: I'm not that fond of mini-jacks. The advantage is that you don't have to worry about polarity and installing is a breeze. But if they become a bit dirty, it is not so easy to clean the inside of the connector on the board and there is no way of tightening for better contact. I'm no where near an expert. But if you ask me, I would prefer screw terminals as long as the wires are solid or with 'eindaderhulzen'. I don't know the English word for that. It is a little tube you clamp around multi threaded wires ;-). 
I hope that makes any sence?

Cheers!
